internal/metamorphic/crossversion/crossversion_test: TestMetaCrossVersion failed

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/7478512?buildTab=log) with [artifacts](https://buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/7478512?buildTab=artifacts#/) on refs/heads/master @ [1bda21f14fc2](https://github.com/cockroachdb/pebble/commits/1bda21f14fc2):


```
=== RUN   TestMetaCrossVersion
    crossversion_test.go:78: Test directory: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/_tmp/366cdd6e86440d4e9e9c98d28c5c6592/TestMetaCrossVersion2899575219/001
    crossversion_test.go:79: Reproduction:
          go test -v -run 'TestMetaCrossVersion' --seed 1668319149902000401 --factor 10 --version crl-release-21.2,f97c66083678,/test-bin/f97c66083678.test --version crl-release-22.1,11d901e8d438,/test-bin/11d901e8d438.test --version crl-release-22.2,966da222263f,/test-bin/966da222263f.test --version master,1bda21f14fc2,/test-bin/1bda21f14fc2.test
        
    crossversion_test.go:97: 0: crl-release-21.2,f97c66083678,/test-bin/f97c66083678.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 1: crl-release-22.1,11d901e8d438,/test-bin/11d901e8d438.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 2: crl-release-22.2,966da222263f,/test-bin/966da222263f.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 3: master,1bda21f14fc2,/test-bin/1bda21f14fc2.test (Mode = -rwxrwxrwx)
    crossversion_test.go:155: Running tests with version f97c6608 with 1 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 11d901e8 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 966da222 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 1bda21f1 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    history.go:181: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/history.go:181
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/history.go:128
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:164
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:103
        	Error:      	Received unexpected error:
        	            	open /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/_tmp/366cdd6e86440d4e9e9c98d28c5c6592/TestMetaCrossVersion2899575219/001/1668319149902000401/1bda21f1_561455885028363443_007/_meta/221113-060040.929/random-000/history: no such file or directory
        	Test:       	TestMetaCrossVersion
--- FAIL: TestMetaCrossVersion (162.89s)
=== RUN   TestMetaCrossVersion/execution-master/sha=966da222-seed=7785165609272340452-opts=random-006(sha=11d901e8-seed=2238823251159598708-opts=random-017(sha=f97c6608-seed=16299888245972413878-opts=random-016(<empty>)))
=== PAUSE TestMetaCrossVersion/execution-master/sha=966da222-seed=7785165609272340452-opts=random-006(sha=11d901e8-seed=2238823251159598708-opts=random-017(sha=f97c6608-seed=16299888245972413878-opts=random-016(<empty>)))
=== CONT  TestMetaCrossVersion/execution-master/sha=966da222-seed=7785165609272340452-opts=random-006(sha=11d901e8-seed=2238823251159598708-opts=random-017(sha=f97c6608-seed=16299888245972413878-opts=random-016(<empty>)))
    crossversion_test.go:221:   Running test with version 1bda21f1 with initial state sha=966da222-seed=7785165609272340452-opts=random-006(sha=11d901e8-seed=2238823251159598708-opts=random-017(sha=f97c6608-seed=16299888245972413878-opts=random-016(<empty>))).
        --- FAIL: TestMetaCrossVersion/execution-master/sha=966da222-seed=7785165609272340452-opts=random-006(sha=11d901e8-seed=2238823251159598708-opts=random-017(sha=f97c6608-seed=16299888245972413878-opts=random-016(<empty>))) (14.88s)
=== RUN   TestMetaCrossVersion/execution-master
    --- FAIL: TestMetaCrossVersion/execution-master (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/7490545?buildTab=log) with [artifacts](https://buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/7490545?buildTab=artifacts#/) on refs/heads/master @ [1bda21f14fc2](https://github.com/cockroachdb/pebble/commits/1bda21f14fc2):


```
=== RUN   TestMetaCrossVersion
    crossversion_test.go:78: Test directory: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/_tmp/366cdd6e86440d4e9e9c98d28c5c6592/TestMetaCrossVersion172883010/001
    crossversion_test.go:79: Reproduction:
          go test -v -run 'TestMetaCrossVersion' --seed 1668416868506911943 --factor 10 --version crl-release-21.2,f97c66083678,/test-bin/f97c66083678.test --version crl-release-22.1,11d901e8d438,/test-bin/11d901e8d438.test --version crl-release-22.2,966da222263f,/test-bin/966da222263f.test --version master,1bda21f14fc2,/test-bin/1bda21f14fc2.test
        
    crossversion_test.go:97: 0: crl-release-21.2,f97c66083678,/test-bin/f97c66083678.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 1: crl-release-22.1,11d901e8d438,/test-bin/11d901e8d438.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 2: crl-release-22.2,966da222263f,/test-bin/966da222263f.test (Mode = -rwxrwxrwx)
    crossversion_test.go:97: 3: master,1bda21f14fc2,/test-bin/1bda21f14fc2.test (Mode = -rwxrwxrwx)
    crossversion_test.go:155: Running tests with version f97c6608 with 1 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 11d901e8 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 966da222 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    crossversion_test.go:155: Running tests with version 1bda21f1 with 10 initial state(s).
=== CONT  TestMetaCrossVersion
    history.go:181: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/history.go:181
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/history.go:128
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:164
        	            				/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test_/crossversion_test.runfiles/com_github_cockroachdb_cockroach/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:103
        	Error:      	Received unexpected error:
        	            	open /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1021/execroot/com_github_cockroachdb_cockroach/_tmp/366cdd6e86440d4e9e9c98d28c5c6592/TestMetaCrossVersion172883010/001/1668416868506911943/1bda21f1_567239179244938813_009/_meta/221114-090859.803/random-000/history: no such file or directory
        	Test:       	TestMetaCrossVersion
--- FAIL: TestMetaCrossVersion (130.15s)
=== RUN   TestMetaCrossVersion/execution-master
    --- FAIL: TestMetaCrossVersion/execution-master (0.00s)
=== RUN   TestMetaCrossVersion/execution-master/sha=966da222-seed=7486251578238709611-opts=random-018(sha=11d901e8-seed=7866569984274576566-opts=standard-019(sha=f97c6608-seed=11239109668367698397-opts=standard-015(<empty>)))
=== PAUSE TestMetaCrossVersion/execution-master/sha=966da222-seed=7486251578238709611-opts=random-018(sha=11d901e8-seed=7866569984274576566-opts=standard-019(sha=f97c6608-seed=11239109668367698397-opts=standard-015(<empty>)))
=== CONT  TestMetaCrossVersion/execution-master/sha=966da222-seed=7486251578238709611-opts=random-018(sha=11d901e8-seed=7866569984274576566-opts=standard-019(sha=f97c6608-seed=11239109668367698397-opts=standard-015(<empty>)))
    crossversion_test.go:221:   Running test with version 1bda21f1 with initial state sha=966da222-seed=7486251578238709611-opts=random-018(sha=11d901e8-seed=7866569984274576566-opts=standard-019(sha=f97c6608-seed=11239109668367698397-opts=standard-015(<empty>))).
        --- FAIL: TestMetaCrossVersion/execution-master/sha=966da222-seed=7486251578238709611-opts=random-018(sha=11d901e8-seed=7866569984274576566-opts=standard-019(sha=f97c6608-seed=11239109668367698397-opts=standard-015(<empty>))) (12.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/internal/metamorphic/meta_test.go
+++ b/internal/metamorphic/meta_test.go
@@ -314,9 +314,10 @@ func TestMeta(t *testing.T) {
 		seed = uint64(time.Now().UnixNano())
 	}
 
-	// Cleanup any previous state.
-	metaDir := filepath.Join(*dir, time.Now().Format("060102-150405.000"))
-	require.NoError(t, os.RemoveAll(metaDir))
+	// Create a directory for test state.
+	require.NoError(t, os.MkdirAll(*dir, 0755))
+	metaDir, err := os.MkdirTemp(*dir, time.Now().Format("060102-150405.000"))
+	require.NoError(t, err)
 	require.NoError(t, os.MkdirAll(metaDir, 0755))
 	defer func() {
 		if !t.Failed() && !*keep {

